tests/test_api_data.py

```diff
@@ -131,3 +131,19 @@ def test_get_parameter_value(
         process_code_deriv=process_code_deriv,
     )
     assert expected == pytest.approx(result)
+
+
+def test_get_dimensions_parameter_code(ptxdata_static):
+    code = ptxdata_static.get_dimensions_parameter_code("country", "Germany")
+    assert code == "DEU"
+
+    code = ptxdata_static.get_dimensions_parameter_code("country", "")
+    assert code == ""
+
+    code = ptxdata_static.get_dimensions_parameter_code("country", None)
+    assert code == ""
+
+    code = ptxdata_static.get_dimensions_parameter_code(
+        "secproc_water", "Specific costs"
+    )
+    assert code == ""

```

Comment:
```suggestion
@pytest.mark.parametrize(
    "dimension, parameter_name, expected_code",
    (
        ("country", "Germany", "DEU"),
        ("country", "", ""),
        ("country", None, ""),
        ("secproc_water", "Specific costs", ""),
    ),
)
def test_get_dimensions_parameter_code(
    ptxdata_static, dimension, parameter_name, expected_code
):
    out_code = ptxdata_static.get_dimensions_parameter_code(dimension, parameter_name)
    assert out_code == expected_code
```

Reply:
You can use mark.parametrize with pytest. This way, each parameter combination gets transformed to a single test function by pytest and it is much easier to find errors if an error is occurring for a parameter combination.